File: domain/interfaces/iPrestamoRepository.go
package interfaces

import "github.com/gdosoftware/biblioteca/domain/modelo"

type IPrestamoRepository interface {
	Create(prestamo *modelo.Prestamo) (*modelo.Prestamo, error)
	Update(id string, prestamo *modelo.Prestamo) (*modelo.Prestamo, error)
	Retrieve(id string) (*modelo.Prestamo, error)
	Delete(id string) error
	FindAll() ([]modelo.Prestamo, error)
}

File: src/main/java/com/gdosoftware/biblioteca/domain/interfaces/IPrestamoRepository.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.gdosoftware.biblioteca.domain.interfaces;

import com.gdosoftware.biblioteca.domain.modelo.Prestamo;
import java.util.List;

/**
 *
 * @author Dani
 */
public interface IPrestamoRepository {
    public Prestamo create(Prestamo libro);
    public Prestamo update(Long id, Prestamo libro);
    public Prestamo retrieve (Long id);
    public void deltete (Long id);
    public List<Prestamo> findAll();
}
